src/main/content/replays/replay-content.ts

```diff
@@ -84,7 +86,11 @@ export class ReplayContentAPI {
         }
     }
 
-    protected async cacheReplay(replayFilePath: string) {
+    protected async cacheReplay(replayFilePath: string, obj: ReplayContentAPI) {
+        // Don't try to cache an empty replay file
+        const stat = await fsPromises.stat(replayFilePath);
+        if (stat.size === 0) return;
```

Comment:
We skip file size if it's 0:
- Will this file caching be re-triggered once the engine finishes running?
- There is still race condition here of engine writing a few bytes but still not full header and this won't be parsable